pagopa/io-app
# chore: [IOPLT-603] Sets Android API Level 34

> [!note]
> Depends on #5864 

## Short description
This PR sets Android API Level to 34 and upgrades `react-native-vision-camera` to solve compatibility issues

## List of changes proposed in this pull request
- Sets Android API Level to 34
- Removes `vision-camera-code-scanner`
- Updates `react-native-vision-camera`
- Refactors the app components using the camera to scan Barcode

## How to test
Test barcode code scanning to check everything is ok on both iOS and Android. Normally use the App on Android to check if the new API Level bump compromises anything
